### PR TITLE
feat: move images from connector-runtime-docker repository

### PR DESCRIPTION
**Pull request by chillleader**
## Description

This PR incorporates Docker images that were previously managed in [connector-runtime-docker](https://github.com/camunda/connector-runtime-docker) repository. I have also adjusted a few things regarding project structure and what images we publish - please take a look below and I'm open to feedback! 🙂 

### Changes to project structure

- New directory is created to manage the minimal viable version the connector runtime: `/connector-runtime/connector-runtime-application`. This is aligned with our modular approach, as we previously shipped the runtime in `spring-boot-starter-camunda-connectors` and in `bundle` versions. Spring Boot starter is not a standalone Java app, and bundle comes with OOTB connectors, so there was no middle ground (viable standalone runtime without any connectors).

- In this PR, the minimal runtime is shipped in 2 forms: Maven module `connector-runtime-application` and Docker image [camunda/connectors](https://hub.docker.com/r/camunda/connectors).

- With this change, the structure of the `/bundle/default-bundle` module was slightly adjusted. It doesn't have its own `@SpringBootApplication` class anymore. Instead, it reuses the one defined in the new module, `connector-runtime-application`. Accordingly, the `src` directory of `default-bundle` is deleted.

### Dockerfiles moved to this repo

- `/connector-runtime/connector-runtime-application/Dockerfile` - base image for connector runtime, no connectors included. [See DockerHub](https://hub.docker.com/r/camunda/connectors).
- `/connector-runtime/connector-runtime-application/example/Dockerfile` - an example Dockerfile that illustrates how one can create a custom connectors bundle in a pure docker setup. ⚠️ **With this PR, we stop publishing this image to Dockerhub, as it duplicates our `default-bundle` image.** We continue to maintain it as an example for custom builds.

### Build pipeline changes

- Added a step to publish SNAPSHOT `camunda/connectors` base image from the new module in our `DEPLOY.md` pipeline
- Added a step to publish versioned + latest tags in `RELEASE.md` pipeline
- Added a step to lint the newly added dockerfile

## Related issues

related to #538

### CODE REVIEW

File: connector-runtime/connector-runtime-application/start.sh [hunk 1]
```diff
@@ -16,4 +16,6 @@ if [[ -n ${DEBUG_JVM_PRINT_JAVA_OPTS} ]]; then
   echo "Applied JVM options: $JAVA_OPTS"
 fi
 
-exec java "${JAVA_OPTS}" -cp '/opt/app/*' io.camunda.connector.bundle.ConnectorRuntimeApplication
+unzip -l /opt/app/connector-runtime-application-0.21.0-SNAPSHOT-with-dependencies.jar | grep ConnectorRuntimeApplication
```
[the remark this answers is not in the document]
Comment (chillleader) [reply connector-runtime/connector-runtime-application/start.sh]: Sorry, this is a debug artifact - I removed this line

[the remark this answers is not in the document]
Comment (chillleader) [reply connector-runtime/connector-runtime-application/start.sh]: @sbuettner please take another look

**Comment by sbuettner** [comment]
@chillleader How would one launch the self-managed version with all connectorswithout building a docker image? Just search for the `https://github.com/camunda/connectors-bundle/blob/538-add-runtime-dockerfiles/connector-runtime/connector-runtime-application/src/main/java/io/camunda/connector/runtime/app/ConnectorRuntimeApplication.java#L23` and execute that under the default-bundle project?

**Comment by chillleader** [comment]
@sbuettner You can still use the docker image from the default-bundle module, as we continue to publish it on [DockerHub](https://hub.docker.com/r/camunda/connectors-bundle). Sorry if it's confusing from the description. I meant that we stop publishing [this image](https://hub.docker.com/r/camunda/connectors-bundle-bash/tags) which was previously built from the [example dockerfile](https://github.com/camunda/connectors-bundle/blob/538-add-runtime-dockerfiles/connector-runtime/connector-runtime-application/example/Dockerfile).

**Comment by sbuettner** [comment]
@chillleader Sorry, I wasnt clear enough. I meant just launching the whole bundle with all dependencies from the IDE.

**Comment by chillleader** [comment]
Yeah then it's exactly like you described, navigate to the main class in `default-bundle` and run it

**Review by sbuettner — APPROVED** [review]

File: connector-runtime/connector-runtime-application/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM eclipse-temurin:17.0.7_7-jre
+
+RUN mkdir /opt/app
+
+COPY start.sh /start.sh
+RUN chmod +x start.sh
+
+# Copy Connector runtime from local build
+COPY target/*-with-dependencies.jar /opt/app/
```
Comment (igpetrov) [inline connector-runtime/connector-runtime-application/Dockerfile:9]: Maybe it even makes sense to simplify it to `*.jar`. WDYT?

File: connector-runtime/connector-runtime-application/example/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ADD https://repo1.maven.org/maven2/io/camunda/connector/connector-sendgrid/${CAMUNDA_CONNECTORS_VERSION}/connector-sendgrid-${CAMUNDA_CONNECTORS_VERSION}-with-dependencies.jar /opt/app/
+ADD https://repo1.maven.org/maven2/io/camunda/connector/connector-slack/${CAMUNDA_CONNECTORS_VERSION}/connector-slack-${CAMUNDA_CONNECTORS_VERSION}-with-dependencies.jar /opt/app/
+ADD https://repo1.maven.org/maven2/io/camunda/connector/connector-sns/${CAMUNDA_CONNECTORS_VERSION}/connector-sns-${CAMUNDA_CONNECTORS_VERSION}-with-dependencies.jar /opt/app/
+ADD https://repo1.maven.org/maven2/io/camunda/connector/connector-sqs/${CAMUNDA_CONNECTORS_VERSION}/connector-sqs-${CAMUNDA_CONNECTORS_VERSION}-with-dependencies.jar /opt/app/
+ADD https://repo1.maven.org/maven2/io/camunda/connector/connector-kafka/${CAMUNDA_CONNECTORS_VERSION}/connector-kafka-${CAMUNDA_CONNECTORS_VERSION}-with-dependencies.jar /opt/app/
+ADD https://repo1.maven.org/maven2/io/camunda/connector/connector-microsoft-teams/${CAMUNDA_CONNECTORS_VERSION}/connector-microsoft-teams-${CAMUNDA_CONNECTORS_VERSION}-with-dependencies.jar /opt/app/
```
Comment (igpetrov) [inline connector-runtime/connector-runtime-application/example/Dockerfile:21]: Feels like we're missing several. E.g., dynamo

**Review by igpetrov — APPROVED** [review]
LGTM, several optional suggestions

File: connector-runtime/connector-runtime-application/Dockerfile (hunk 2 above)
Comment (chillleader) [reply connector-runtime/connector-runtime-application/Dockerfile:9]: This way the non-executable slim jar will also make it to the docker image.I think it makes sense to keep the classpath clean and only add the needed files

**Review by sbuettner — APPROVED** [review]

**Pull request merged**